core/src/main/java/org/infinispan/persistence/PersistenceUtil.java

```diff
@@ -160,11 +161,15 @@ public static Path getQualifiedLocation(GlobalConfiguration globalConfiguration,
    }
 
    public static Path getLocation(GlobalConfiguration globalConfiguration, String location) {
-      Path persistentLocation = Paths.get(globalConfiguration.globalState().persistentLocation());
+      GlobalStateConfiguration globalState = globalConfiguration.globalState();
+      Path persistentLocation = Paths.get(globalState.persistentLocation());
       if (location == null)
          return persistentLocation;
 
       Path path = Paths.get(location);
+      if (!globalState.enabled()) {
+          return path;
+      }
```

Comment:
I think we need this:

```java
public static Path getLocation(GlobalConfiguration globalConfiguration, String location) {
      GlobalStateConfiguration globalState = globalConfiguration.globalState();
      Path persistentLocation = Paths.get(globalState.persistentLocation());
      if (location == null) {
         if (!globalConfiguration.globalState().enabled())
            throw Log.CONFIG.storeLocationRequired();
         return persistentLocation;
      }

      Path path = Paths.get(location);
      if (path.isAbsolute()) {
         // Ensure that the path lives under the global persistent location
         if (path.startsWith(persistentLocation)) {
            return path;
         } else {
            throw PERSISTENCE.forbiddenStoreLocation(path, persistentLocation);
         }
      }
      return persistentLocation.resolve(path);
   }
```

Reply:
That change removes the essence of this fix, i.e. preventing a PersistenceException when global state is disabled.
However, I think your point is that we should handle the case where global state is disabled *and* the location is null within in this method.  That we can do.